File: Leetcode/firstBadVersion.java
public int firstBadVersion(int n) {
        int i = 1;
        int j = n;
        while (i <= j) {
            int median =  i + (j - i) / 2;
            if (isBadVersion(median)) {
                j = median - 1;
            } else {
                i = median + 1;
            }
        }
        return i;
    }

File: Leetcode/Python/firstBadVersion.py
def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        i,j = 1, n
        while i <= j:
            middle = (i + j)/2
            if isBadVersion(middle):
                j = middle - 1
            else:
                i = middle + 1
        return i
